src/server/server.cc

```diff
@@ -1753,3 +1755,50 @@ std::list<std::pair<std::string, uint32_t>> Server::GetSlaveHostAndPort() {
   slave_threads_mu_.unlock();
   return result;
 }
+
+// The numeric cursor consists of a 32-bit hash, a 16-bit time stamp, and a 16-bit counter, with the highest bit set to
+// 1 to prevent a zero cursor from occurring. The hash is used to prevent information leakage. The time_stamp is used to
+// prevent the generation of the same cursor in the extremely short period before and after a restart.
+static uint64_t GetNumberCursor(const std::string &key_name, uint16_t counter) {
+  auto hash = static_cast<uint32_t>(std::hash<std::string>{}(key_name));
+  auto time_stamp = static_cast<uint16_t>(util::GetTimeStamp());
+  // set first bit 1, so number cursor not be 0
+  return static_cast<uint64_t>(hash) | static_cast<uint64_t>(time_stamp) << 32 | static_cast<uint64_t>(counter) << 48 |
+         static_cast<uint64_t>(1) << 63;
+}
+
+std::string Server::GenerateCursorFromKeyName(const std::string &key_name, const char *prefix) {
+  if (!config_->redis_cursor_compatible) {
+    // add prefix for SCAN
+    return prefix + key_name;
+  }
+  auto counter = cursor_counter_.fetch_add(1);
```

Comment:
So prefix name will not be added in new-cursor?

Reply:
[Prefix will be auto remove in ParseCursor.](https://github.com/apache/incubator-kvrocks/blob/unstable/src/commands/scan_base.h#L57-L64
)  If we add prefix, we will need to remove it again in the `Execute` function, which I think is redundant.
I think prefix is just a readability identifier.